internal/semver/semver.go

```diff
@@ -130,39 +142,76 @@ func (v *Version) Compare(other *Version) int {
 		return 1
 	}
 	// prerelease number (e.g. "alpha1" vs "alpha2")
-	if v.PrereleaseNumber < other.PrereleaseNumber {
+	// Note: Lack of prerelease number is considered lower precedence than
+	// the same prerelease when it has a number - https://semver.org/#spec-item-11.
+	if v.PrereleaseNumber == nil && other.PrereleaseNumber != nil {
 		return -1
 	}
-	if v.PrereleaseNumber > other.PrereleaseNumber {
+	if v.PrereleaseNumber != nil && other.PrereleaseNumber == nil {
 		return 1
 	}
+	if v.PrereleaseNumber != nil && other.PrereleaseNumber != nil {
+		if *v.PrereleaseNumber < *other.PrereleaseNumber {
+			return -1
+		}
+		if *v.PrereleaseNumber > *other.PrereleaseNumber {
+			return 1
+		}
+	}
 	return 0
 }
 
 // String formats a Version struct into a string.
 func (v *Version) String() string {
 	version := fmt.Sprintf("%d.%d.%d", v.Major, v.Minor, v.Patch)
 	if v.Prerelease != "" {
-		version += "-" + v.Prerelease + v.PrereleaseSeparator + v.PrereleaseNumber
+		version += "-" + v.Prerelease
+		if v.PrereleaseNumber != nil {
+			version += v.PrereleaseSeparator + strconv.Itoa(*v.PrereleaseNumber)
+		}
 	}
 	return version
 }
 
 // incrementPrerelease increments the pre-release version number, or appends
 // one if it doesn't exist.
-func (v *Version) incrementPrerelease() error {
-	if v.PrereleaseNumber == "" {
+func (v *Version) incrementPrerelease() {
+	if v.PrereleaseNumber == nil {
 		v.PrereleaseSeparator = "."
-		v.PrereleaseNumber = "1"
-		return nil
+		v.PrereleaseNumber = new(int)
```

Comment:
nit: Perhaps add a comment to this block along the lines of  "If PrereleaseNumber is nil, it's initialized to 0 and then incremented to 1." It is was not immediate obvious to me.

Reply:
Good call. PTAL!